### PR TITLE
Update EmailPreference model

### PR DESCRIPTION
**Pull request by sureshc**
Don't set `opt_in` to false if user previously agreed to be added to our mailing list.  Use a constant array to specify the list of permitted values for `source`

### CODE REVIEW

File: dashboard/app/models/email_preference.rb [hunk 1]
```diff
@@ -22,14 +22,25 @@ class EmailPreference < ApplicationRecord
   validates_uniqueness_of :email
   validates_email_format_of :email
 
+  SOURCE_TYPES = [
+    REGISTRATION = 'Teacher account sign up',
```
Comment (breville) [inline dashboard/app/models/email_preference.rb]: I wonder if these should be more like `"teacher_account_sign_up"` so they're a little more formal and less organic, for querying purposes?

File: dashboard/test/models/email_preference_test.rb [hunk 2]
```diff
@@ -7,19 +7,48 @@ class EmailPreferenceTest < ActiveSupport::TestCase
...
+    email_preference = create :email_preference, opt_in: true
+    EmailPreference.upsert!(
+      email: email_preference.email,
+      opt_in: false,
+      ip_address: '172.16.6.1',
+      source: EmailPreference::REGISTRATION,
+      form_kind: 'form text version 3.14159'
+    )
+    email_preference.reload
+    assert_equal email_preference.opt_in, true
```
Comment (breville) [inline dashboard/test/models/email_preference_test.rb]: good testing!

File: dashboard/app/models/email_preference.rb (hunk 1 above)
Comment (sureshc) [reply dashboard/app/models/email_preference.rb]: I was thinking that that constant (`EmailPreference::REGISTRATION`) provides the formal identifier of a source type, which can be used by our code and reporting, but the value of each constant is a piece of text that the user will understand, if they ask or we show them, how we got their consent.

Comment (breville) [reply dashboard/app/models/email_preference.rb]: We will need to loc all such strings, so being humanly readable in code isn't important from that point of view.

**Pull request merged**